csrc/root_domain_map.cpp

```diff
@@ -230,28 +230,24 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
     // Producers:
     //   query = [N, H, L, E]
     //   key = [N, H, S, E]
-    //   value = [N, H, S, Ev]
+    //   value = [N, H, S, E]
     // Consumers:
-    //   output = [N, H, L, Ev]
+    //   output = [N, H, L, E]
     //   logsumexp = [N, H, L]
     //   cum_seq_q/k = [N + 1]
 
-    // Map N, H from any input (query/key/value)
+    // Map N, H, E from any input (query/key/value)
     for (auto idx : c10::irange(consumer_root.size())) {
-      if (idx < 2) {
+      // Map N, H, E from any input to output
+      if (idx != 2) {
```

Comment:
But this part looks like we are mapping all those as the same.

Reply:
NVM. I see that `updatePairwiseRootDomainMap` checks for broadcast inside the lambda.